### ISSUE TITLE
New check `replace-if-with-loop`

### ISSUE THREAD

**Issue by clavedeluna** [PyCQA/pylint#7876]
### Current problem

bad:
```
if len(nums) > 0:
	for num in nums:
		print(num)
else:
	print("no nums!")

if nums:
	for num in nums:
		print(num)
else:
	print("no nums!")

```

good:

```
for num in nums:
	print("inside for")
	print(x)
else:
	print("no nums!")
```

invalid:
```
for x in []:
	print("inside for")
	print(x)
elif: # invalid!
	print("inside else")	
```

### Desired solution

Have pylint suggest not checking presence or length of an iterator and instead start iterating directly.

Suggested name so far: `replace-if-with-loop`, but could probably come up with something more meaningful.

**Comment by cdce8p** [comment]
Your "good" example won't work, unfortunately.
```py
nums = [1, 2, 3]
for num in nums:
    print("inside for")
    print(num)
else:
    print("no nums!")
```
```
inside for
1
inside for
2
inside for
3
no nums!
```

The `else` on a for loop is executed whenever the loop exits normally, i.e. without a `break`, and not only if no elements are looped over.
https://book.pythontips.com/en/latest/for_-_else.html#else-clause

**Comment by clavedeluna** [comment]
Good point! I always forget that detail. I'll update the examples to remove the else, I think it's still valid without it.

**Comment by orSolocate** [comment]
+1 for the idea!

Do we want to raise the message only when the `for` loop is the first node inside the `if` node? I Other cases could be False Positives..

Should the checker look for `len(nums) > 0` and `if nums` only or are there other cases we want to tackle? Examples that comes to my mind is `if nums is not None` and `if num != []` these could be replaced with `if nums`, but still could be present in the code..

**Comment by clavedeluna** [comment]
> Do we want to raise the message only when the for loop is the first node inside the if node? I Other cases could be False Positives..

Yes. If node then for node, nothing in between I think, to avoid FP.

```
if len(nums) > 0:
    var = 0
    for num in nums:
        print(num + var)
       var += 1
```
Is an example that we _could_ catch but maybe that's a future iteration of this checker.

>  Examples that comes to my mind  ...

Yes. I think it's reasonable that the first iteration also checks for these test cases.

```
if nums:
    for num in nums:
        print(num)

if nums is not None:
    for num in nums:
        print(num)

if nums != []:
    for num in nums:
        print(num)

```

**Comment by cdce8p** [comment]
> For `if nums is not None:` None is not iterable, you're going to get a TypeError if you remove the check.

You could use `for num in nums or ()` instead.
